src/mui/input/ReferenceInput.js

```diff
@@ -157,15 +169,21 @@ export class ReferenceInput extends Component {
             resource,
             label,
             source,
-            referenceRecord,
             allowEmpty,
             matchingReferences,
             basePath,
             onChange,
             children,
             meta,
         } = this.props;
-        if (!referenceRecord && !allowEmpty) {
```

Comment:
Same remark as for the `ReferenceArrayInput`, the app musn't break if the `referenceRecord` doesn't exist.